### DISCUSSION TITLE
Converting PEM to JWK in V3

### DISCUSSION THREAD

**Discussion by justin0108** [panva/jose#149] [Q&A] [answered]
I am migrating from V2 to V3.
Previously in V2 I am able to use `JWK.asKey` to convert the file content to JWK.
In V3 api, I cannot find such API anymore.
`parseJWK` requires an object but all I have is the PEM contents.

**Comment by panva** [comment, marked as answer]
[crypto.createPublicKey](https://nodejs.org/api/crypto.html#crypto_crypto_createpublickey_key) or [crypto.createPrivateKey](https://nodejs.org/api/crypto.html#crypto_crypto_createprivatekey_key) to end up with `KeyObject` then use [jose/jwk/from_key_like](https://github.com/panva/jose/blob/v3.5.1/docs/functions/_jwk_from_key_like_.fromkeylike.md#readme) to get a JWK.

**Comment by justin0108** [comment]
Thank you so much for the prompt reply.
I have tested its working fine according to your suggestion.
In case anybody face the same issue, below is the code snippet I used

```
const pemPublicKey = Buffer.from(pemPublicKeyBase64String, 'base64').toString();
const cryptoKeyObj = crypto.createPublicKey(pemPublicKey);
const jwk = await fromKeyLike(cryptoKeyObj);
const rsaPublicKey = await parseJwk(jwk, 'RS256');
```

Another thing to note is the node version.
You need minimum version of node `12.19.0`.
Originally I was facing issue of importing the module as I was using `12.14.0`

**Reply by panva** [reply]
> You need minimum version of node 12.19.0.
> Originally I was facing issue of importing the module as I was using 12.14.0

That's documented in the readme.

**Reply by justin0108** [reply]
Yup that's right. I miss that out and proceed to try the library.
So I pointed it here again in case anybody stumble the same issue with importing.
Initially I didn't expect the node version will cause importing issue.
Thanks

**Reply by rambabusaravanan** [reply]
@panva  How do I use this in browser.

I receive a rsa public key string and I'll need to create JWE. Trying to parse JWK from string but `crypto.createPublicKey` is available only in nodejs api and not in browser api. Even it's not available in 'crypto-browserify'

**Reply by panva** [reply]
https://github.com/panva/jose/blob/v3.11.3/docs/types/types.keylike.md


use https://developer.mozilla.org/en-US/docs/Web/API/SubtleCrypto/importKey

**Reply by rambabusaravanan** [reply]
Thanks a lot @panva for very quick reply
